src/components/RegistrationForm/RegistrationForm.tsx

```diff
@@ -0,0 +1,198 @@
+      confirmPassword: true,
+    });
+
+    const isValid =
+      validators.name(formData.firstName) &&
+      validators.name(formData.lastName) &&
+      validators.email(formData.email) &&
+      validators.password(formData.password) &&
+      formData.password === formData.confirmPassword;
+
+    if (!isValid) {
+      setShakeAnimation(true);
+      setTimeout(() => setShakeAnimation(false), 500);
+
+      return;
+    }
+
+    setIsSubmitted(true);
+    setFormData(INITIAL_FORM_DATA);
+    setTouched(INITIAL_TOUCHED);
+
+    // eslint-disable-next-line no-console
+    console.log('Form submitted:', formData);
+  };
+
+  return (
+    <div>
+      <form
+        onSubmit={handleSubmit}
+        className={`${style.registrationFormCard} ${shakeAnimation ? style.shakeAnimation : ''}`}
+      >
+        <div className={style.inputSection}>
+          <h1 className={style.registrationTitle}>register to buk klab</h1>
+
+          <FormInput
+            type="text"
+            name="firstName"
+            placeholder="First Name"
+            value={formData.firstName}
+            error={formErrors.firstName}
+            errorMessage={ERROR_MESSAGES.firstName}
+            onChange={handleChange}
+            onBlur={() => handleBlur('firstName')}
+          />
+
+          <FormInput
+            type="text"
+            name="lastName"
+            placeholder="Last Name"
+            value={formData.lastName}
+            error={formErrors.lastName}
+            errorMessage={ERROR_MESSAGES.lastName}
+            onChange={handleChange}
+            onBlur={() => handleBlur('lastName')}
+          />
+
+          <FormInput
+            type="email"
+            name="email"
+            placeholder="Email"
+            value={formData.email}
+            error={formErrors.email}
+            errorMessage={ERROR_MESSAGES.email}
+            onChange={handleChange}
+            onBlur={() => handleBlur('email')}
+          />
+
+          <FormInput
+            type="password"
+            name="password"
+            placeholder="Choose your password"
+            value={formData.password}
+            error={formErrors.password}
+            errorMessage={ERROR_MESSAGES.password}
+            onChange={handleChange}
+            onBlur={() => handleBlur('password')}
+          />
+
+          <FormInput
+            type="password"
+            name="confirmPassword"
+            placeholder="Confirm your password"
+            value={formData.confirmPassword}
+            error={formErrors.confirmPassword}
+            errorMessage={ERROR_MESSAGES.passwordMatch}
+            onChange={handleChange}
+            onBlur={() => handleBlur('confirmPassword')}
+          />
+
+          {isSubmitted ? (
+            <div className={style.successMessage}>Registration successful!</div>
+          ) : (
+            <button type="submit" className={style.registerButton}>
+              register
+            </button>
+          )}
```

Comment:
I would expect to be redirected upon submit. Therefore no `isSubmitted` is needed in that case. I leave this decision up to your design